gum/arch-arm64/gumarm64writer.c

```diff
@@ -707,6 +707,29 @@ gum_arm64_writer_put_add_reg_reg_imm (GumArm64Writer * self,
 }
 
 void
+gum_arm64_writer_put_add_reg_reg_reg (GumArm64Writer * self,
+                                      GumArm64Reg dst_reg,
+                                      GumArm64Reg left_reg,
+                                      GumArm64Reg right_reg)
+{
+  GumArm64RegInfo rd, rl, rr;
+
+  gum_arm64_writer_describe_reg (self, dst_reg, &rd);
+  gum_arm64_writer_describe_reg (self, left_reg, &rl);
+  gum_arm64_writer_describe_reg (self, right_reg, &rr);
+
+  g_assert_cmpuint (rd.width, ==, rl.width);
+  g_assert_cmpuint (rd.width, ==, rr.width);
+
+  guint32 flags = 0;
+  if(rd.width == 64) {
+    flags |= 0x8000000;
+  }
+
+  gum_arm64_writer_put_instruction (self, rd.sf | 0xb000000 | flags | rd.index | (rl.index << 5) | (rr.index << 16));
```

Comment:
Would be great to break the line at 80 cols (see style elsewhere in this file). Note that `frida-core` doesn't have an 80 col limit, as the plan is to move `frida-gum` to the same conventions later; but for now let's try to keep things consistent within each module.